DigitalLearningSolutions.Data/DataServices/SupervisorService.cs

```diff
@@ -1063,17 +1070,76 @@ public IEnumerable<CandidateAssessmentSupervisorVerificationSummary> GetCandidat
 
         public int IsSupervisorDelegateExistAndReturnId(int? supervisorAdminId, string delegateEmail, int centreId)
         {
-            int? delegateUserId = (int?)connection.ExecuteScalar(
-                 @"SELECT da.UserID AS DelegateUserID 
+            int? delegateUserId = null;
+
+            var user = connection.Query<UserAccount>(
+                @$"SELECT u.ID, u.PrimaryEmail, u.FirstName, u.LastName,  u.Active
                             FROM Users u
-                            INNER JOIN DelegateAccounts da
-                            ON da.UserID = u.ID
-	                        LEFT JOIN UserCentreDetails ucd
+                            LEFT JOIN UserCentreDetails ucd
 	                        ON ucd.UserID = u.ID
-                            AND ucd.CentreID = da.CentreID
-                            WHERE (ucd.Email = @delegateEmail OR u.PrimaryEmail = @delegateEmail)
-                            AND u.Active = 1 
-                            AND da.CentreID = @centreId", new { delegateEmail, centreId });
+                            WHERE (u.PrimaryEmail = @delegateEmail OR ucd.Email = @delegateEmail)
+                            AND u.Active = 1", new { delegateEmail }).FirstOrDefault();
+            if (user != null)
+            {
+                delegateUserId = (int?)connection.ExecuteScalar(
+                     @"SELECT Top 1 da.UserID AS DelegateUserID 
+                            FROM DelegateAccounts da
+                            LEFT JOIN UserCentreDetails ucd
+	                        ON ucd.UserID = da.UserID
+                            WHERE (da.UserID = @Id OR ucd.Email = @delegateEmail) AND da.CentreID = @centreId
+                            AND da.Active = 1", new { user?.Id, delegateEmail, centreId });
+
+                if (delegateUserId == null)
+                {
+                    var initials = (user.FirstName.Substring(0, 1) +
+                                    user.LastName.Substring(0, 1)).ToUpper();
+
+                    var candidateNumber = connection.QueryFirst<string>(

```

Comment:
This looks to be copied and pasted code. I think it would be safer to move this to a common method, instead and call it from both places.

Reply:
Created common method to generate candidate number.